libs/vkd3d/device.c

```diff
@@ -449,6 +449,7 @@ static const struct vkd3d_shader_quirk_info f1_2020_quirks = {
     NULL, 0, VKD3D_SHADER_QUIRK_FORCE_TGSM_BARRIERS,
 };
 
+/* Checks if the end of the application's name matches. */
```

Comment:
I would instead add a compare mode field to the vkd3d_shader_quirk_meta, where we either go for exact or "ends with" matching. Would also allows us to add "begins with" and "contains" later if necessary.